### ISSUE TITLE
ggez/rustc nightly regression

### ISSUE THREAD

**Issue by U007D** [ggez/ggez#1148]
**Describe the bug**
Simple white square on black canvas demo app fails on nightly `rustc` (tested on 1.67.0-2022-12-05 and 1.67.0-2022-12-08), yet works on stable (1.65.0).

**To Reproduce**
1. Clone and `cargo run` [this project](https://github.com/U007D/ggez_test)
2. Note the window contains a white square
3. Change `rust-toolchain.toml` to `channel = "nightly"`
4. `cargo run` again
5. Note the window no longer contains the white square

**Expected behavior**
Window should contain the white square.

**Hardware and Software:**
 - ggez version: 0.8.1
 - OS: Mac OS Ventura 13.0.1
 - Graphics card: Radeon Pro 560X 4GB & Intel UHD Graphics 630 1536MB
 - Graphics card drivers: 
 ```
 Radeon Pro 560X:

  Chipset Model:	Radeon Pro 560X
  Type:	GPU
  Bus:	PCIe
  PCIe Lane Width:	x8
  VRAM (Total):	4 GB
  Vendor:	AMD (0x1002)
  Device ID:	0x67ef
  Revision ID:	0x00c2
  ROM Revision:	113-C980AL-075
  VBIOS Version:	113-C97501U-005
  EFI Driver Version:	01.A1.075
  Automatic Graphics Switching:	Supported
  gMux Version:	5.0.0
  Metal Support:	Metal 2
  Displays:
Color LCD:
  Display Type:	Built-In Retina LCD
  Resolution:	2880 x 1800 Retina
  Framebuffer Depth:	30-Bit Color (ARGB2101010)
  Mirror:	On
  Mirror Status:	Hardware Mirror
  Online:	Yes
  Automatically Adjust Brightness:	No
  Connection Type:	Internal
S32D850:
  Resolution:	2560 x 1440 (QHD/WQHD - Wide Quad High Definition)
  UI Looks like:	2560 x 1440 @ 60.00Hz
  Framebuffer Depth:	30-Bit Color (ARGB2101010)
  Main Display:	Yes
  Mirror:	On
  Mirror Status:	Master Mirror
  Online:	Yes
  Rotation:	Supported
  Connection Type:	DVI or HDMI
  Adapter Firmware Version:	7.55
```

**Comment by nobbele** [comment]
can you try devel branch

**Comment by JulianKnodt** [comment]
I also ran into this bug (for the circle not the rectangle though) when starting with the generative art tutorial, and found that the `devel` branch fixed it.

**Comment by chamons** [comment]
I can confirm this on a project I'm working on.

 I ran into this with:

```
[toolchain]
channel = "nightly-2022-12-21"
targets = [ "x86_64-pc-windows-msvc" ]
```

and swapping to devel branch fixed it for me. 

Unfortunately bevy requires nightly, so I'm forced to use devel.

**Comment by chamons** [comment]
I did some bisecting, and 

channel = "nightly-2022-10-21"- WORKS
channel = "nightly-2022-11-21" - WORKS
channel = "nightly-2022-12-01" - FAILS

**Comment by locriacyber** [comment]
No idea why this is happening.  All mesh drawing functions fail.

**Comment by chamons** [comment]
I am now seeing this with stable, which is significantly more annoying, as I'd need devel (and thus a custom ggez) for all platforms (or pin to an older rust). 

Would it be possible to get devel as a crate, even as a beta?

**Comment by chamons** [comment]
I wonder if somehow it broke due to https://github.com/rust-lang/rust/pull/102750/

**Comment by locriacyber** [comment]
> I am now seeing this with stable, which is significantly more annoying, as I'd need devel (and thus a custom ggez) for all platforms (or pin to an older rust).
> 
> Would it be possible to get devel as a crate, even as a beta?

You can pin the crate version with `{ git = "xxxxx.git", branch = "devel" }`

**Comment by chamons** [comment]
@locriacyber - That worked for me, thanks!

The invocation I used in my Cargo.toml, in case others hit this is:

```
ggez = "0.8.1"

[patch.crates-io]
ggez = { git = 'https://github.com/ggez/ggez.git', branch = "devel"  }
```

**Comment by Vixeliz** [comment]
This appears to be fixed as of 0.9.0-rc unless I'm mistaken lmk if I was wrong in closing the issue and i'll reopen.
